nene/math.nelua

```diff
@@ -10,53 +10,65 @@ global Math.Vec2 = @record{
 local Vec2 = @Math.Vec2
 ## Vec2.value.is_vec = true
 
-global Vec2.zero: Vec2 = { x = 0, y = 0}
-
-function Vec2.__eq(l: Vec2, r: Vec2): boolean
-  return l.x == r.x and l.y == r.y
+function Vec2.__eq(a: Vec2, b: Vec2): boolean
+  return a.x == b.x and a.y == b.y
 end
 
-function Vec2.__len(v: Vec2): number
-  return math.sqrt( (v.x * v.x) + (v.y * v.y) )
+function Vec2.__add(a: Vec2, b: Vec2): Vec2
+  return (@Vec2){ x = a.x + b.x, y = a.y + b.y }
 end
 
-function Vec2.__add(l: Vec2, r: Vec2): Vec2
-  return (@Vec2){ x = l.x + r.x, y = l.y + r.y }
+function Vec2.__sub(a: Vec2, b: Vec2): Vec2
+  return (@Vec2){ x = a.x - b.x, y = a.y - b.y }
 end
 
-function Vec2.__sub(l: Vec2, r: Vec2): Vec2
-  return (@Vec2){ x = l.x - r.x, y = l.y - r.y }
-end
+function Vec2.__mul(a: overload(Vec2, number), b: overload(Vec2, number)): Vec2
+  -- one of the arguments must be a Vec2
+  ## staticassert(a.type.is_vec or b.type.is_vec, 'neither types is a Vec2')
```

Comment:
This will usually never happen unless `Vec2.__mul` is called directly, that is, the compiler will only call the methamethod if at least one of the types is a Vec2.

Reply:
> This will usually never happen unless Vec2.__mul is called directly,

True, it doesn't makes much sense to add an assert to such strange situation, I'll remove the `staticassert`. Thanks for the feedback ^^.